shared/constants/unlock-folders.js

```diff
@@ -0,0 +1,31 @@
+/* @flow */
+
+import HiddenString from '../util/hidden-string'
+import type {TypedAction} from '../constants/types/flux'
+import type {DeviceID} from '../constants/types/flow-types'
+
+export type Device = {
+  type: 'desktop' | 'mobile',
+  name: string,
+  deviceID: DeviceID
+}
+
+export const loadDevices = 'unlockFolders:loadDevices'
+export type LoadDevices = TypedAction<'unlockFolders:loadDevices', {devices: Array<Device>}, {}>
+
+// transistions to the next paper key phase
+export const toPaperKeyInput = 'unlockFolders:toPaperKeyInput'
+export type ToPaperKeyInput = TypedAction<'unlockFolders:toPaperKeyInput', {}, {}>
+
+// TODO: not sure if we even need this, we might be able to do it all without having to store the paperkey
```

Comment:
we can avoid having the keys travel through redux entirely using the dynamic routes with a onSumbit being plumbed through directly to a response.result() call